lib/datautils/owid/datautils/dataframes.py

```diff
@@ -632,6 +639,14 @@ def combine_two_overlapping_dataframes(
     # Give priority to df1 on overlapping values.
     combined, df2 = df1.align(df2)
 
+    new_columns = df2.columns.difference(df1.columns)
+    for col in new_columns:
+        try:
+            combined[col] = combined[col].astype(df2[col].dtype, copy=False)
```

Comment:
The issue here is that if there is a column in `df2` that is not present in `df1`, it gets introduced in `combined` as an all-NaN column with float dtype (that's what `align` does). 
And then in the next step when doing `combined.fillna(df2)`, it will fill those NaN values with the values of the column in `df2`. But if that column is a string column, pandas will use the "common dtype" for the resulting column, and for a combination of `float64` and `string` dtype, that will be `object` dtype, losing the `string` dtype of the column in `df2`.

As long as strings were stored in a `object` dtype column, this didn't matter that the result was object dtype. But now with a proper string dtype, I think ideally that dtype gets preserved.